dataline-workers/src/main/java/io/dataline/workers/singer/SingerSyncWorker.java

```diff
@@ -68,16 +70,23 @@ public DefaultSyncWorker(TapFactory<SingerMessage> singerTapFactory,
 
     final SingerMessageTracker singerMessageTracker = new SingerMessageTracker();
 
-    try (Stream<SingerMessage> tap = singerTapFactory.create(tapConfig, jobRoot);
-        CloseableConsumer<SingerMessage> consumer = singerTargetFactory.create(targetConfig, jobRoot)) {
+    try {
+      singerTarget.start(targetConfig, jobRoot);
+      singerTap.start(tapConfig, jobRoot);
 
-      tap.takeWhile(record -> !cancelled.get()).peek(singerMessageTracker).forEach(consumer);
+      while (!cancelled.get() && singerTap.hasNext()) {
+        SingerMessage message = singerTap.next();
+        singerMessageTracker.accept(message);
+        singerTarget.consume(message);
+      }
 
+      singerTap.stop();
+      singerTarget.stop();
     } catch (Exception e) {
-      LOGGER.debug(
-          "Sync worker failed. Tap error log: {}.\n Target error log: {}",
-          IOs.readFile(jobRoot, TAP_ERR_LOG),
-          IOs.readFile(jobRoot, TARGET_ERR_LOG));
+      LOGGER.error("Sync worker failed. Tap error log: {}.\n Target error log: {}",
+          Files.exists(jobRoot.resolve(TAP_ERR_LOG)) ? IOs.readFile(jobRoot, TAP_ERR_LOG) : "<null>",
```

Comment:
Exception where not displayed because they happened before the files were created